quantized_mesh_tile/tile_stitcher.py

```diff
@@ -0,0 +1,419 @@
+        size = len(self._side_vertices.values())
+        return EdgeConnection.BOTH_SIDES == size
+
+    @property
+    def is_broken_on_center(self):
+        return self.is_broken_on(self.edge_info)
+
+    @property
+    def is_broken_on_neighbour(self):
+        return self.is_broken_on('c')
+
+    def is_broken_on(self, edge_side):
+        # type: (str) -> bool
+        size = len(self._side_vertices.values())
+        return EdgeConnection.ONE_SIDE == size and self.is_side(edge_side)
+
+    def is_side(self, edge_side):
+        # type: (str) -> bool
+        """
+
+        :param edge_side: the participating side of the edge ('w','n','e','s')
+        :return: Returns True, if a vertex of the given side is registered in this
+        connection
+        """
+        return edge_side in self._side_vertices.keys()
+
+
+class TileStitcher(object):
+    """
+        The worker class to stitch terrain files together
+
+        Constructor arguments:
+
+        ''center_tile''
+            the the center tile, from which the neighbouring edges are stitched,
+            if neighbour tiles are added
+
+
+        Usage example::
+        import os
+        from quantized_mesh_tile.tile_stitcher import TileStitcher, load_tile,
+                                                        get_neighbours_south_east
+
+        directory_base_path = '/data/terrain/'
+        levels = [4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16]
+
+        #walk through level and file hierarchy
+        for level in levels:
+            directory_path = os.path.join(directory_base_path, str(level))
+            terrain_files = []
+            for root, dirs, files in os.walk(directory_path, topdown=True):
+                for name in files:
+                    candidate_path = os.path.join(root, name)
+                    if candidate_path.endswith('.terrain'):
+                        terrain_files.append(candidate_path)
+
+            for tile_path in terrain_files:
+
+                y = int(os.path.basename(tile_path).split('.')[0])
+                x = int(os.path.basename(os.path.dirname(tile_path)))
+                print('processing {0} ...'.format(tile_path))
+                neighbours = get_neighbours_south_east(level, x, y)
+                center_tile = tile_stitcher.load_tile(tile_path, x, y, level)
+
+                stitcher = TileStitcher(center_tile)
+                for n, tile_info in neighbours.items():
+                    n_z, n_x, n_y = tile_info
+
+                    neighbour_path = os.path.join(directory_path,
+                                    '%s/%s.terrain' % (n_x, n_y))
+                    if os.path.exists(neighbour_path):
+                        print("\tadding Neighbour {0}...".format(neighbour_path))
+                        tile = tile_stitcher.load_tile(neighbour_path, n_x, n_y, level)
+                        stitcher.add_neighbour(tile)
+                stitcher.stitch_together()
+                stitcher.save()
```

Comment:
that's a lot of code... to stich

Reply:
Do you mean the usage of the tile_stitcher?

Reply:
Yes, do think we can hide some of the complexity?

Reply:
Yes, of course, here is everything changable... the reason for the complexity was the wish to offer flexibility for different usecases or patterns of batchprocessing.  I was unsure if a seperate save() and saveTo() is needed... but as everthing else in this PR it must fit into your style/design and concept